src/components/PopupWithForm.js

```diff
@@ -28,9 +28,18 @@ export default class PopupWithForm extends Popup {
         return values;
     }
 
-    _submitForm = () => {
+    _submitForm = (event) => {
+        event.preventDefault(); // Prevent the form from submitting normally
+        this.setLoading(true); // Disable the submit button and show loading text
         const inputValues = this._getInputValues();
-        this._handleFormSubmit(inputValues);
+        this._handleFormSubmit(inputValues, () => {
+            this.setLoading(false); // Re-enable the submit button after the form is handled
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.


So, you can't use both actions (closing and `this.setLoading(false)`) in one callback.